### DISCUSSION TITLE
Test Filters : json query

### DISCUSSION THREAD

**Discussion by nikhilp79** [thunderclient/thunder-client-support#1107] [Q&A] [answered]
I am validating an api response that has a format as shown below: 
```json
{
	"results": [
		{
			"id": 3,
			"name": "a"
		},
		{
			"id": 4,
			"name": "d"
		},
		{
			"id": 3,
			"name": "Z",
			"size": {
				"length": 5,
				"width": 4
			}
		}
	]
}
```
"size" object node is not present on all array elements. I am writing a test (using Built-in Filters) to give me the "length" of first element where "size" object is present. I was hoping this would work : " json.results | filter(size > 0) | at(0).size.length"  "equal" 5  . But is failing. Any idea why? It looks like "json.results | filter(size > 0)| at(0)" is giving me expected results. Or is there a better approach to test this response?

**Comment by rangav** [comment, marked as answer]
Hi @nikhilp79 You dont need built in filter for this.

Use this expression in tests left input for `json query` - `json.results[size>0][0].size.length`

<img width="764" alt="Screenshot 2023-04-07 at 20 53 12" src="https://user-images.githubusercontent.com/8637550/230669152-c6155383-eeb8-45d2-9569-6470dd7df102.png">

Docs here
https://github.com/rangav/thunder-client-support/blob/master/docs/testing.md#test-arrays
